modules/sievefilters/modules.php

```diff
@@ -1458,18 +1478,17 @@ function get_classic_filter_modal_content()
 if (!hm_exists('get_mailbox_filters')) {
     function get_mailbox_filters($mailbox, $site_config, $user_config, $html=false)
     {
-        try {
-            $factory = get_sieve_client_factory($site_config);
-            $client = $factory->init($user_config, $mailbox);
-            $scripts = [];
-            foreach ($client->listScripts() as $script) {
-                if (strstr($script, 'cypht')) {
-                    $scripts[] = $script;
-                }
-            }
-        } catch (PhpSieveManager\Exceptions\SocketException $e) {
+        $factory = get_sieve_client_factory($site_config);
+        $client = $factory->init($user_config, $mailbox);
+        if (!$client) {
             return '';
         }
+        $scripts = [];
+        foreach ($client->listScripts() as $script) {
```

Comment:
Methods on $client can still throw exceptions - e.g. listScripts, calls sendCommand which calls readResponse which can throw a SocketException. We should either refactor php-sieve-manager or handle these exceptions here - otherwise, Cypht will break when unhandled exceptions occur.

Reply:
@kroky Now Client exceptions are caught